### PR TITLE
Added LED support for nRF52840 Dongle

### PR DESCRIPTION
**Pull request by paTelshREya**
Added LED support for nRF52840 Dongle.
- identify the state of the node
  - Leader: red
  - Router: blue
  - Chile: green
- indicate when a message is published

Branch: shreya/led-support

### CODE REVIEW

File: prj.conf [hunk 1]
```diff
@@ -38,7 +38,7 @@ CONFIG_OPENTHREAD_MANUAL_START=n
 CONFIG_OPENTHREAD_UPTIME=y
 
 # MQTT-SNCLIENT
-CONFIG_MQTT_SNCLIENT_TOPIC_PREFIX="ot/alex"
+CONFIG_MQTT_SNCLIENT_TOPIC_PREFIX="ot/Shreya"
```
Comment (ajlennon) [inline prj.conf]: Let's not change this - I probably shouldn't have mine in here either

File: src/gpio.c [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    }
+	if (err < 0) {
+		LOG_WRN("LED set ON failed");
+	}
+}
+
+void otLedToggle(uint8_t aLed) {
+	int err = 0;
+	switch (aLed) {
+        case 1:
```
Comment (ajlennon) [inline src/gpio.c]: Let's add in some defines for the LED number and the colour

File: src/gpio.c [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    }
+	if (err < 0) {
+		LOG_WRN("LED toggle failed");
+	}
+}
+
+void otLedBlink(uint8_t aLed) {
+    switch (aLed) {
+        case 1:
+            otLedToggle(1);
```
Comment (ajlennon) [inline src/gpio.c]: I don't think we should have sleeps in these functions. Sleeps can block important things from happening so let's not to this. Probably best taking the function out if it is blocking

File: src/gpio.h [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+#include "openthread/instance.h"
+#include "openthread/thread.h"
+
+// Definitions
+
+/* 1000 msec = 1 sec */
+#define SLEEP_TIME_MS   300
+
+/* The devicetree node identifier for the "led" alias. */
+#define LED0_NODE DT_ALIAS(led0)
```
Comment (ajlennon) [inline src/gpio.h]: Let's use the DT aliases for the LED nodes - it helps to know what LED and what colour we are trying to change

**Review by ajlennon — CHANGES_REQUESTED** [review]

File: src/gpio.h (hunk 4 above)
Comment (paTelshREya) [reply src/gpio.h]: ok

File: prj.conf (hunk 1 above)
Comment (paTelshREya) [reply prj.conf]: forgot that!

File: src/gpio.c (hunk 3 above)
Comment (paTelshREya) [reply src/gpio.c]: I'll remove the Blink function!

**Comment by ajlennon** [comment]
Sorry to be a pain - maybe I misunderstood you but it would be better to have the JSON formatting on a separate branch. Then we could merge that in straight away while we continue to look at the LED support

**Review by ajlennon — APPROVED** [review]
Let's get this in

**Pull request merged**